pkg/apimachinery/registered/registered.go

```diff
@@ -42,75 +57,101 @@ var (
 	// enabledVersions represents all enabled API versions. It should be a
 	// subset of registeredVersions. Please call EnableVersions() to add
 	// enabled versions.
-	enabledVersions = map[unversioned.GroupVersion]struct{}{}
+	enabledVersions map[unversioned.GroupVersion]struct{}
 
 	// map of group meta for all groups.
-	groupMetaMap = map[string]*apimachinery.GroupMeta{}
+	groupMetaMap map[string]*apimachinery.GroupMeta
 
 	// envRequestedVersions represents the versions requested via the
 	// KUBE_API_VERSIONS environment variable. The install package of each group
 	// checks this list before add their versions to the latest package and
 	// Scheme.  This list is small and order matters, so represent as a slice
-	envRequestedVersions = []unversioned.GroupVersion{}
-)
-
-func init() {
-	loadKubeAPIVersions()
+	envRequestedVersions []unversioned.GroupVersion
 }
 
-func loadKubeAPIVersions() {
-	// Env var KUBE_API_VERSIONS is a comma separated list of API versions that
-	// should be registered in the scheme.
-	kubeAPIVersions := os.Getenv("KUBE_API_VERSIONS")
+func NewManager(kubeAPIVersions string) (*Manager, error) {
+	m := &Manager{
+		registeredVersions:      map[unversioned.GroupVersion]struct{}{},
+		thirdPartyGroupVersions: []unversioned.GroupVersion{},
+		enabledVersions:         map[unversioned.GroupVersion]struct{}{},
+		groupMetaMap:            map[string]*apimachinery.GroupMeta{},
+		envRequestedVersions:    []unversioned.GroupVersion{},
+	}
+
 	if len(kubeAPIVersions) != 0 {
 		for _, version := range strings.Split(kubeAPIVersions, ",") {
 			gv, err := unversioned.ParseGroupVersion(version)
 			if err != nil {
-				glog.Fatalf("invalid api version: %s in KUBE_API_VERSIONS: %s.",
-					version, os.Getenv("KUBE_API_VERSIONS"))
+				return nil, fmt.Errorf("invalid api version: %s in KUBE_API_VERSIONS: %s.",
+					version, kubeAPIVersions)
 			}
-			envRequestedVersions = append(envRequestedVersions, gv)
+			m.envRequestedVersions = append(m.envRequestedVersions, gv)
 		}
 	}
+	return m, nil
 }
 
 // Resets everything to clean room for the start of a test
+// TODO: delete this
```

Comment:
Would be good to add details of why we want to delete this.

Perhaps say that testing should use its own instance of Manager and once we do that this should no longer be required.